### ISSUE TITLE
MUITheme needs to affect Login as it does for the rest of the Application

### ISSUE THREAD

**Issue by zifnab87** [marmelab/react-admin#2368]
**Is your feature request related to a problem? Please describe.**
Adding a Mui theme on `<Admin>` using `theme={customTheme}` doesn't affect Login

![image](https://user-images.githubusercontent.com/526062/46259846-de8b6680-c4e6-11e8-9997-8faade430e58.png)

![image](https://user-images.githubusercontent.com/526062/46259855-f662ea80-c4e6-11e8-96b6-7f1a191c948d.png)

**Describe the solution you'd like**
Adding a MUI theme like that would affect Login as it does with the rest of the Application: 
```
const customTheme = createMuiTheme({
    palette: {
        primary: {
            light: '#ffcc00',
            main: '#ff9900',
            dark: "#cc6600",
            contrastText: '#ffffff',
        },
        secondary: {
            light: '#ffcc00',
            main: '#ff9900',
            dark: "#cc6600",
            contrastText: '#ffffff',
        },
    },
});
```
```
<Admin
        theme={customTheme} [...] >
     [....]
</Admin>
```
Something that looks like:
![image](https://user-images.githubusercontent.com/526062/46259860-15fa1300-c4e7-11e8-9133-7e61bb9e3def.png)

**Describe alternatives you've considered**
Using withStyles feels hacky and it is difficult to change error color, labels when inputs are on focus etc

```
import React from 'react';
import { Login, LoginForm } from 'react-admin';
import { withStyles } from '@material-ui/core';
import customTheme from '../theme';

const styles = ({
    main: { background: '#333' },
    avatar: {
        background: 'url(//cdn.example.com/background.jpg)',
        backgroundRepeat: 'no-repeat',
        backgroundSize: 'contain',
        height: 80,
    },
    icon: { display: 'none' },
});

const CustomLoginForm = withStyles({
    button: { 
        background: customTheme.palette.primary.light, 
        '&:hover': { background: customTheme.palette.primary.main } 
    },
})(LoginForm);

const CustomLoginPage = props => (
    <Login
        loginForm={<CustomLoginForm theme={customTheme}/>}
        {...props}
    />
);

export default withStyles(styles)(CustomLoginPage);
```
```
<Admin
        loginPage={CustomLoginPage} [...] > 
    [...] 
</Admin>
```

**Comment by trsteel88** [comment]
Is there a solution to bring the theme through that is defined in app.js?

I tried using what's in the demo (https://github.com/marmelab/react-admin/blob/master/examples/demo/src/Login.js) but it doesn't match my app theme.

**Comment by Kmaschta** [comment]
Fixed by #2591

**Comment by trsteel88** [comment]
If I don't set a theme, the login page still comes through as pink (and the logged in areas with a pink header).

Am I missing something?
